src/lightning/pytorch/strategies/xla.py

```diff
@@ -157,14 +150,11 @@ def model_to_device(self) -> None:
         self.model = self.wrapped_model.to(self.root_device)
 
     def barrier(self, name: Optional[str] = None, *args: Any, **kwargs: Any) -> None:
-        if self.is_distributed:
-            import torch_xla.core.xla_model as xm
+        import torch_xla.core.xla_model as xm
 
-            xm.rendezvous(name)
+        xm.rendezvous(name)
```

Comment:
Have you checked that this works with world_size=1? There must have been a reason we added this condition, and it may have been resolved in later xla versions.

Reply:
There's no motivation in https://github.com/Lightning-AI/lightning/pull/7241

But with world_size=1, `SingleTPUStrategy` is used instead.

Still, I tried it in case somebody passes `strategy="xla", devices=1` and it's working. But I found an issue when no barrier name is passed:

```python
  File "examples/pytorch/bug_report/bug_report_model.py", line 30, in training_step
    self.trainer.strategy.barrier()
  File "/home/carmocca/lightning/src/lightning/pytorch/strategies/xla.py", line 155, in barrier
    xm.rendezvous(name)
  File "/usr/local/lib/python3.8/dist-packages/torch_xla/core/xla_model.py", line 1112, in rendezvous
    return torch_xla._XLAC._xla_rendezvous(get_ordinal(), tag, payload, replicas)
TypeError: _xla_rendezvous(): incompatible function arguments. The following argument types are supported:
    1. (arg0: int, arg1: str, arg2: str, arg3: List[int]) -> List[bytes]

Invoked with: 0, None, b'', []
```